services/ircd/default.nix

```
@@ -0,0 +1,406 @@
+        name = "Redbrick Intersocs Server IRC admin";
+        email = "admins@${tld}";
+      };
+
+      bind = [
+        {
+          address = ipAddress;
+          port = "6697";
+          ssl = "openssl";
+        }
+        {
+          address = ipAddress;
+          port = "6667";
+          ssl = "openssl";
+          type = "servers";
+        }
+      ];
+
+      connect = [
+        {
+          name="Secure";
+          parent="Main";
+          port="6697";
+        }
+        {
+          name = "Main";
+          allow ="*";
+          hash="sha256";
+          timeout="10";
+          limit="5000";
+        }
+      ];
+      cidr = {
+        ipv4clone="32";
+        ipv6clone="128";
+      };
+      class = [
+        {
+          name="Shutdown";
+          commands="DIE RESTART REHASH LOADMODULE UNLOADMODULE RELOADMODULE GLOADMODULE GUNLOADMODULE GRELOADMODULE";
+          privs="users/auspex channels/auspex servers/auspex users/mass-message channels/high-join-limit users/flood/no-throttle users/flood/increased-buffers";
+          usermodes="*";
+          chanmodes="*";
+        }
+        {
+          name="SACommands";
+          commands="SAJOIN SAPART SANICK SAQUIT SATOPIC SAKICK SAMODE OJOIN";
+        }
+        {
+          name="ServerLink";
+          commands="CONNECT SQUIT RCONNECT RSQUIT MKPASSWD ALLTIME SWHOIS JUMPSERVER LOCKSERV UNLOCKSERV";
+          usermodes="*";
+          chanmodes="*";
+          privs="servers/auspex";
+        }
+        {
+          name="BanControl";
+          commands="KILL GLINE KLINE ZLINE QLINE ELINE TLINE RLINE CHECK NICKLOCK NICKUNLOCK SHUN CLONES CBAN CLOSE";
+          usermodes="*";
+          chanmodes="*";
+        }
+        {
+          name="OperChat";
+          commands="WALLOPS GLOBOPS";
+          usermodes="*";
+          chanmodes="*";
+          privs="users/mass-message";
+        }
+        {
+          name="HostCloak";
+          commands="SETHOST SETIDENT SETIDLE CHGNAME CHGHOST CHGIDENT";
+          usermodes="*";
+          chanmodes="*";
+          privs="users/auspex";
+        }
+      ];
+      type = [
+        {
+          name="NetAdmin";
+          classes="SACommands OperChat BanControl HostCloak Shutdown ServerLink";
+          modes="+s +cCqQ";
+        }
+      ];
+      oper = [
+        {
+          name="butlerx";
+          hash="hmac-sha256";
+          # password: A hash of the password (see above option) hashed
+          # with /mkpasswd <hash> <password>. See m_password_hash in modules.conf
+          # for more information about password hashing.
+          password=/var/secrts/ircd/butlerx.sha256.pass;
+          host="*@*";
+          sslonly="yes";
+          type="NetAdmin";
+        }
+      ];
+      files = {
+        motd=./motd.txt;
+        rules=./rules.txt;
+      };
+      channels = {
+        users = "80";
+        opers = "600";
+      };
+      dns = [
+        {
+          server = "136.206.15.26";
```

Avoid hard coding this.
```suggestion
          server = builtins.head config.networking.nameservers;
```

And on line 1 add `{ config, ... }:`

---

didn't know this existed